### PR TITLE
13.0 fix misc las

### PR DESCRIPTION
**Pull request by smetl**
**[IMP] account: Remove dead code**

This field has been introduced by 2d1b8e9
but is no longer used anywhere and is not a valid field on account.move.line.

**[FIX] account: Fix lost constraint during the account-pocalypse**

A journal entry must always have the same currency as the account when
a secondary currency is set on this one.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -2985,6 +2979,13 @@ def _compute_tax_audit(self):
     # CONSTRAINT METHODS
     # -------------------------------------------------------------------------
 
+    @api.constrains('currency_id', 'account_id')
+    def _check_account_currency(self):
+        for line in self:
+            account_currency = line.account_id.currency_id
+            if account_currency and account_currency != line.company_currency_id and account_currency != line.currency_id:
+                raise ValidationError(_('The selected account of your Journal Entry forces to provide a secondary currency. You should remove the secondary currency on the account.'))
```
Comment (oco-odoo) [inline addons/account/models/account_move.py]: The account selected on your journal entry .....

(also change in translation, of course :p )

**Comment by oco-odoo** [comment]
@robodoo delegate+

**Comment by smetl** [comment]
@robodoo r+

**Comment by oco-odoo** [comment]
@robodoo rebase-ff

**Pull request closed without merging**